homeassistant/components/climate/reproduce_state.py

```diff
@@ -41,7 +41,7 @@ async def call_service(service: str, keys: Iterable, data=None):
         data = data or {}
         data["entity_id"] = state.entity_id
         for key in keys:
-            if key in state.attributes:
+            if key in state.attributes and state.attributes[key] is not None:
```

Comment:
The following code would use one lookup instead of three:

```python
   if (value := state.attributes.get(key)) is not None:
     data[key] = value
```

Reply:
Done.